nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ListDomainsCommand.java

```diff
@@ -60,99 +64,107 @@
     boolean header;
-    
+
     /*
      * Override the validate method because super.validate() calls initDomain,
      * and since we don't have a domain name yet, we aren't ready to call that.
      */
     @Override
-    protected void validate()
-            throws CommandException, CommandValidationException {
+    protected void validate() throws CommandException, CommandValidationException {
     }
 
     @Override
     protected int executeCommand() throws CommandException, CommandValidationException {
         try {
-            File domainsDirFile = ok(domainDirParam)
-                    ? new File(domainDirParam) : DomainDirs.getDefaultDomainsDir();
-
+            File domainsDirFile = ok(domainDirParam) ? new File(domainDirParam) : DomainDirs.getDefaultDomainsDir();
             DomainConfig domainConfig = new DomainConfig(null, domainsDirFile.getAbsolutePath());
             DomainsManager manager = new PEDomainsManager();
             String[] domainsList = manager.listDomains(domainConfig);
             programOpts.setInteractive(false);  // no prompting for passwords
             if (domainsList.length > 0) {
                 if (longOpt) {
                     String headings[] = {"DOMAIN", "ADMIN_HOST", "ADMIN_PORT", "RUNNING", "RESTART_REQUIRED"};
-                    ColumnFormatter cf = header ? new ColumnFormatter(headings) : 
-                            new ColumnFormatter();
+                    ColumnFormatter cf = header ? new ColumnFormatter(headings)
+                            : new ColumnFormatter();
                     for (String dn : domainsList) {
                         DomainInfo di = getStatus(dn);
-                        cf.addRow(new Object[] {
-                            dn, 
-                            di.adminAddr.getHost(), 
-                            di.adminAddr.getPort(), 
+                        cf.addRow(new Object[]{
+                            dn,
+                            di.adminAddr.getHost(),
+                            di.adminAddr.getPort(),
                             di.status,
-                            di.restartRequired                              
+                            di.restartRequired
                         });
                     }
                     logger.info(cf.toString());
                 } else {
                     for (String dn : domainsList) {
                         logger.info(getStatus(dn).statusMsg);
                     }
-                }               
-            }
-            else {
+                }
+            } else {
                 logger.fine(strings.get("NoDomainsToList"));
             }
-        }
-        catch (Exception ex) {
+        } catch (Exception ex) {
             throw new CommandException(ex.getLocalizedMessage());
         }
         return 0;
     }
 
+    protected List<String> getRunningDomains() throws IOException, DomainException, CommandException {
```

Comment:
I would probably have this method return something more generic. Maybe a Map<String, String> mapping domains to their status or something similar. IMHO, the list-domains command shouldn't care about whether the domain is running or not. It's just me being pedantic, but I think it's better to invert the control on this one

Reply:
Right, makes sense. I shall bow to the pedantry in this case ;)